pkg/pubsub/adapter/converters/scheduler.go

```diff
@@ -20,65 +20,41 @@ import (
 	"context"
 	"errors"
 
-	cloudevents "github.com/cloudevents/sdk-go"
-	. "github.com/cloudevents/sdk-go/pkg/cloudevents"
-	cepubsub "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub"
-	pubsubcontext "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub/context"
-
-	"github.com/google/knative-gcp/pkg/apis/events/v1alpha1"
+	"github.com/google/knative-gcp/pkg/apis/events/v1beta1"
 	"github.com/google/knative-gcp/pkg/reconciler/events/scheduler/resources"
-)
 
-const (
-	CloudSchedulerConverter = "com.google.cloud.scheduler"
+	"cloud.google.com/go/pubsub"
+	cev2 "github.com/cloudevents/sdk-go/v2"
 )
 
-func convertCloudScheduler(ctx context.Context, msg *cepubsub.Message, sendMode ModeType) (*cloudevents.Event, error) {
-	tx := pubsubcontext.TransportContextFrom(ctx)
-	// Make a new event and convert the message payload.
-	event := cloudevents.NewEvent(cloudevents.VersionV1)
-	event.SetID(tx.ID)
-	event.SetTime(tx.PublishTime)
-	// We do not know the content type and we do not want to inspect the payload,
-	// thus we set this generic one.
-	event.SetDataContentType("application/octet-stream")
-	event.SetType(v1alpha1.CloudSchedulerSourceExecute)
-	// Set the schema if it comes as an attribute.
-	if val, ok := msg.Attributes["schema"]; ok {
-		delete(msg.Attributes, "schema")
-		event.SetDataSchema(val)
-	}
+func convertCloudScheduler(ctx context.Context, msg *pubsub.Message) (*cev2.Event, error) {
+	event := cev2.NewEvent(cev2.VersionV1)
+	event.SetID(msg.ID)
+	event.SetTime(msg.PublishTime)
+	event.SetType(v1beta1.CloudSchedulerSourceExecute)
+
 	// Set the source and subject if it comes as an attribute.
-	jobName, ok := msg.Attributes[v1alpha1.CloudSchedulerSourceJobName]
+	// We added this attributes so that we could identify the scheduler.
+	// We should remove them here.
+	jobName, ok := msg.Attributes[v1beta1.CloudSchedulerSourceJobName]
 	if ok {
-		delete(msg.Attributes, v1alpha1.CloudSchedulerSourceJobName)
+		delete(msg.Attributes, v1beta1.CloudSchedulerSourceJobName)
```

Comment:
I think my comment below was ambiguous, I was thinking that the deletion of these attributes is no longer necessary.

Reply:
ohh I think they are, because with them we populate ce-source and ce-subject...

Reply:
darn, I get your point... no need to delete them argggg

Reply:
the arguments are necessary, no need to delete, let me remove that

Reply:
Done, thanks for the catch